LadybugTools_Adapter/Convert/Constructions/EnergyMaterial.cs

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.LadybugTools;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BH.Adapter.LadybugTools
+{
+    public static partial class Convert
+    {
+        public static BH.oM.LadybugTools.EnergyMaterial ToEnergyMaterial(Dictionary<string, object> oldObject)
+        {
+            if (Enum.TryParse((string)oldObject["roughness"], out Roughness roughness))
+            {
+                try
+                {
+                    return new oM.LadybugTools.EnergyMaterial()
+                    {
+                        Name = (string)oldObject["identifier"],
+                        Thickness = (double)oldObject["thickness"],
+                        Conductivity = (double)oldObject["conductivity"],
+                        Density = (double)oldObject["density"],
+                        SpecificHeat = (double)oldObject["specific_heat"],
+                        Roughness = roughness,
+                        ThermalAbsorptance = (double)oldObject["thermal_absorptance"],
+                        SolarAbsorptance = (double)oldObject["solar_absorptance"],
+                        VisibleAbsorptance = (double)oldObject["visible_absorptance"]
+                    };
```

Comment:
It will expand the method somewhat, but I would have these being all their own try-catches, and a suitable default used if they can't convert the type successfully. That way, we might get some of the data even if we can't get all of it